src/Inventory.zig

```diff
@@ -1124,18 +1125,17 @@ pub const Command = struct { // MARK: Command
 				.hand => .{.hand = try reader.readInt(u32)},
 				.recipe => .{
 					.recipe = blk: {
-						var itemList = main.List(struct {amount: u16, item: *const main.items.BaseItem}).initCapacity(main.stackAllocator, len);
+						var itemList = main.List(struct {amount: u16, item: BaseItemIndex}).initCapacity(main.stackAllocator, len);
 						defer itemList.deinit();
 						while(reader.remaining.len >= 2) {
 							const resultAmount = try reader.readInt(u16);
 							const itemId = try reader.readUntilDelimiter(0);
-							const resultItem = main.items.getByID(itemId) orelse return error.Invalid;
-							itemList.append(.{.amount = resultAmount, .item = resultItem});
+							itemList.append(.{.amount = resultAmount, .item = BaseItemIndex.fromId(itemId) orelse return error.Invalid});
 						}
 						if(itemList.items.len != len) return error.Invalid;
 						// Find the recipe in our list:
 						outer: for(main.items.recipes()) |*recipe| {
-							if(recipe.resultAmount == itemList.items[0].amount and recipe.resultItem == itemList.items[0].item and recipe.sourceItems.len == itemList.items.len - 1) {
+							if(recipe.resultAmount == itemList.items[0].amount and recipe.resultItem.index == itemList.items[0].item.index and recipe.sourceItems.len == itemList.items.len - 1) {
```

Comment:
Make it a packed struct, then you don't need to access the index to compare.

Reply:
Yeee, I did that after few comparisons, I must have missed this one, maybe some others.